### PR TITLE
Handle sliced structs properly in pack/contiguous_split.

### PR DESCRIPTION
**Pull request by nvdbaranec**
Pre-sliced struct columns (those with a positive offset) were not being handled correctly.

Dependent PR has been merged.

### CODE REVIEW

**Review by ttnghia — CHANGES_REQUESTED** [review]

File: cpp/include/cudf/detail/null_mask.cuh [hunk 1]
```diff
@@ -141,6 +142,285 @@ void inplace_bitmask_binop(
...
+  CHECK_CUDA(stream.value());
+
+  // third, adjust counts in segment boundaries (if segments are not
+  // word-aligned)
+
+  constexpr size_type block_size{256};
+
+  cudf::detail::grid_1d grid(num_ranges, block_size);
+
+  subtract_set_bits_range_boundaries_kerenel<<<grid.num_blocks,
```
[the remark this answers is not in the document]
Comment (nvdbaranec) [reply cpp/include/cudf/detail/null_mask.cuh:360]: Wait, what happened here?  Didn't I fix this yesterday?

[the remark this answers is not in the document]
Comment (ttnghia) [reply cpp/include/cudf/detail/null_mask.cuh:360]: Yesterday typo was in your other PR.

File: cpp/include/cudf/detail/null_mask.cuh [hunk 2]
```diff
@@ -141,6 +142,285 @@ void inplace_bitmask_binop(
...
+ *
+ * @param[in] bitmask The bitmask whose non-zero bits outside the range in the
+ * boundary words will be counted.
+ * @param[in] num_ranges The number of ranges
+ * @param[in] first_bit_indices The indices (inclusive) of the first bit in each
+ * range
+ * @param[in] last_bit_indices The indices (exclusive) of the last bit in each
+ * range
+ * @param[in,out] null_counts The number of non-zero bits in each range to be
+ * updated
```
Comment (ttnghia) [inline cpp/include/cudf/detail/null_mask.cuh:164]: Sorry that I'm a bit OCD about the style. This is just a suggestion (align multiple-line param doxygen with the parameter name), you can ignore it if you don't like :smiley: 
```suggestion
 * @param[in] bitmask The bitmask whose non-zero bits outside the range in the
 *                    boundary words will be counted.
 * @param[in] num_ranges The number of ranges
 * @param[in] first_bit_indices The indices (inclusive) of the first bit in each
 *                     range
 * @param[in] last_bit_indices The indices (exclusive) of the last bit in each
 *                     range
 * @param[in,out] null_counts The number of non-zero bits in each range to be
 *                          updated
```

File: cpp/include/cudf/detail/null_mask.cuh [hunk 3]
```diff
@@ -141,6 +142,285 @@ void inplace_bitmask_binop(
...
+  {
+    auto bit_index = _d_bit_indices[i];
+    return word_index(bit_index) + ((_inclusive || intra_word_index(bit_index) == 0) ? 0 : 1);
+  }
+};
+
+/**
+ * @brief Functor that returns the number of set bits for a specified word
+ * of a bitmask array.
+ *
```
Comment (ttnghia) [inline cpp/include/cudf/detail/null_mask.cuh:240]:
```suggestion
```

File: cpp/include/cudf/detail/null_mask.cuh [hunk 4]
```diff
@@ -141,6 +142,285 @@ void inplace_bitmask_binop(
...
+
+  /**
+   * @brief Constructor of a functor that converts bit indices to bitmask word
+   * indices.
+   *
+   * @param[in] inclusive Flag that indicates whether bit indices are inclusive
+   * or exclusive.
+   * @param[in] d_bit_indices Pointer to an array of bit indices
+   */
+  __host__ to_word_index(bool inclusive, size_type const* d_bit_indices)
```
Comment (ttnghia) [inline cpp/include/cudf/detail/null_mask.cuh:225]: I'm not sure if we ever need the `__host__` qualifier here.

**Review by ttnghia — APPROVED** [review]

**Review by robertmaynard — APPROVED** [review]
CMake changes LGTM

**Review by vuule — APPROVED** [review]

**Comment by nvdbaranec** [comment]
Temporarily adding do-not-merge tag so @charlesbluca can run some additional tests.

**Review by vuule — APPROVED** [review]
still looks good :)

**Review by ttnghia — APPROVED** [review]

**Comment by nvdbaranec** [comment]
@gpucibot merge

**Pull request merged**